django/template/base.py

```diff
@@ -393,28 +393,34 @@ def create_token(self, token_string, position, lineno, in_tag):
 
 
 class DebugLexer(Lexer):
+    # This parallels the use of tag_re.split() in Lexer.tokenize().
+    def _tag_re_split(self):
+        prev_end = 0
+        for match in tag_re.finditer(self.template_string):
+            start, end = match.span()
+            yield (prev_end, start)
+            yield (start, end)
+            prev_end = end
+        yield (prev_end, len(self.template_string))
+
     def tokenize(self):
         """
         Split a template string into tokens and annotates each token with its
         start and end position in the source. This is slower than the default
         lexer so only use it when debug is True.
         """
+        # For maintainability, it is helpful if the implementation below can
+        # continue to closely parallel Lexer.tokenize()'s implementation.
+        template_string = self.template_string
+        in_tag = False
         lineno = 1
         result = []
-        upto = 0
-        for match in tag_re.finditer(self.template_string):
-            start, end = match.span()
-            if start > upto:
-                token_string = self.template_string[upto:start]
-                result.append(self.create_token(token_string, (upto, start), lineno, in_tag=False))
+        for position in self._tag_re_split():
+            token_string = template_string[slice(*position)]
+            if token_string:
```

Comment:
I found the name `bit` made it harder to understand what `Lexer.tokenize()` was doing when I was trying to understand this code, so I added a separate commit instead to change `bit` to `token_string` in `Lexer`. Now they will match. I think the name `token_string` is an improvement because now it matches the name of the corresponding parameter in [`Lexer.create_token()`](https://github.com/django/django/blob/8208381ba6a3d1613bb746617062ccf1a6a28591/django/template/base.py#L366), which is what the variable is used for. It's also more descriptive and parallels the `template_string` attribute, which is used in the method. `bit` on the other hand doesn't really have any specific meaning (or it sounds like a boolean 0/1, which was confusing).